third_party/amd/backend/compiler.py

```diff
@@ -360,7 +371,10 @@ def make_amdgcn(src, metadata, options):
 
     @staticmethod
     def make_hsaco(src, metadata, options):
-        hsaco = amd.assemble_amdgcn(src, options.arch, '')
+        target_features = ''
+        if os.environ.get("TRITON_ENABLE_ADDRESS_SANITIZER", "0") == "1":
+            target_features = '+xnack'
+        hsaco = amd.assemble_amdgcn(src, options.arch, target_features)
```

Comment:
Why do you need to include the +xnack again here?

Reply:
The target feature needs to be passed in to both the LLVM IR layer and the AMDGCN backend. I also thought it would be picked up if I just included it at the IR level but removing it here causes it to fail to filter down.